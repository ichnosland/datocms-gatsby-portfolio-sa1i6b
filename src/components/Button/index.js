/**
*
* Button
*
*/

import styled, { css } from 'styled-components';
//import { lighten, darken, transparentize } from 'polished';
import media from 'style/mediainjector';
import { colore, transparent } from 'style/color';
import Svg from '../Svg';

export const ButtonStyle = css`
  display: inline-flex;
  align-items: center;
  justify-content: center;
  appearance: none;
  font-family: ${(props) => props.theme.fonts.buttons.fontFamily};
  font-weight: ${(props) => props.theme.fonts.buttons.fontWeight};
  ${({ full }) => full && 'width: 100% !important;'};
  ${({ half }) => half && 'width: 50% !important;'};
  color: ${(props) => props.color ? props.color : colore.ui.contrast};
  text-align: center;
  margin: ${(props) => props.margin};
  padding: ${(props) => props.padding ? props.padding : '0.4em  1.4em'};
  border: ${(props) => props.border ? props.border : '1px solid transparent'};
  background-color: ${(props) => props.bgcolor ? props.bgcolor : props.theme.brand};
  border-radius: ${(props) => props.radius ? props.radius : props.theme.radius.buttons};
  &:disabled {
    background-color: ${(props) => lighten(0.3, (props.bgcolor ? props.bgcolor : props.theme.brand))};
    color: ${(props) => transparentize(0.5, (props.color ? props.color : colore.ui.contrast))};
    pointer-events: none;
    svg {
        opacity: .3;
      }
  }
  ${({ round }) => round && css`
    padding: 0;
    width: ${(props) => props.width};
    height: ${(props) => props.width};
    border-radius: 100%;
  `}
  ${({ actioncolor }) => actioncolor && css`
    background-color: ${colore.actions[actioncolor]};
    &:disabled {
      background-color: ${lighten(0.3, colore.actions[actioncolor])};
      color: ${lighten(0.1, colore.actions[actioncolor])};
    }
  `}
  ${({ admin }) => admin && css`
    color: ${(props) => props.theme.brand};
    border: 1px solid ${colore.ui.neutralBg};
    background-color: ${colore.ui.contrast};
    &:disabled {
      background-color: ${darken(0.1, colore.ui.contrast)};
      color: ${(props) => lighten(0.4, props.theme.brand)};
    }
    &:active {
      background-color: ${darken(0.1, colore.ui.contrast)};
      border-color: ${darken(0.1, colore.ui.neutralBg)};
    }
  `}
  ${({ outline }) => outline && css`
    color: ${(props) => props.theme.brand};
    border: 1px solid ${(props) => props.theme.brand};
    background-color: ${colore.ui.contrast};
    &:disabled {
      background-color: ${darken(0.1, colore.ui.contrast)};
      color: ${(props) => lighten(0.4, props.theme.brand)};
    }
    &:active {
      background-color: ${darken(0.1, colore.ui.contrast)};
      border-color: ${darken(0.1, colore.ui.neutralBg)};
    }
    svg {
      fill: ${(props) => props.fill ? props.fill : props.theme.brand};
    }
  `}
  ${({ bone }) => bone && css`
    color: ${colore.ui.contrast};
    border: 2px solid ${colore.ui.contrast};
    background-color: ${transparent};
    &:hover,
    &:active {
      background-color: rgba(0,0,0,.1);
    }
  `}
  &:active,
  &:focus {
    outline: none;
  }
  ${(props) => props.standard && css`
    min-width: 180px;
    max-width: 320px;
  `};
  ${({ shadow }) => shadow && css`
    border: none;
    box-shadow: 0 0 4px 1px rgba(0,0,0,0.20);
  `}
  ${media.iphone45`
    ${({ standard }) => standard && 'min-width: 140px;'}
    padding: ${(props) => props.padding ? props.padding : '0.4em  0.6em;'};
    svg {
      margin: 0 6px 0 0;
    }
  `}
`;

export const Button = styled.button`
  ${ButtonStyle}
`;

Button.defaultProps = {
  theme: {
    brand: '#00BBEF',
    radius: {
      buttons: '6px',
    },
    fonts: {
      buttons: {
        fontFamily: 'inherit',
        fontWeight: 'inherit',
      },
    },
  },
  actioncolor: '',
};

export const SubmitButton = styled.input`
  ${ButtonStyle}
`;

SubmitButton.defaultProps = {
  theme: {
    brand: '#00BBEF',
    radius: {
      buttons: '6px',
    },
    fonts: {
      buttons: {
        fontFamily: 'inherit',
        fontWeight: 'inherit',
      },
    },
  },
  actioncolor: '',
};

export const ActionButtonStyle = css`
  font-size: 21px;
  padding: 21px 30px;
  background-color: ${(props) => props.bgcolor};
  border: ${(props) => props.border ? props.border : '2px solid white'};
  color: ${(props) => props.color ? props.color : colore.ui.contrast};
  font-weight: 600;
  ${({ shadow }) => shadow && css`
    border: none;
    box-shadow: 0 0 4px 1px rgba(0,0,0,0.20);
  `}
  ${({ sectionbutton }) => sectionbutton && css`
    min-width: 50%;
    font-size: 18px;
    padding: 14px 30px;
    ${media.lt667`
      width: 100%;
    `}
  ${media.lt480`
    font-size: 16px;
    `}
  `}
  &:disabled {
    background-color: ${(props) => lighten(0.3, (props.bgcolor ? props.bgcolor : colore.landing.green))};
    color: ${(props) => transparentize(0.5, (props.bgcolor ? props.bgcolor : colore.ui.contrast))};
  }
`;

export const ActionButton = styled(Button)`
  ${ActionButtonStyle};
`;

export const Icon = styled(Svg)`
  width: ${(props) => props.size ? props.size : '18px'};
  height: ${(props) => props.size ? props.size : '18px'};
  margin: ${(props) => props.margin};
  fill: ${(props) => props.fill ? props.fill : colore.ui.contrast};
  ${({ right }) => right && css`
    margin: 0 -6px 0 6px;
  `}
  ${({ left }) => left && css`
    margin: 0 6px 0 -6px;
  `}
  ${({ brandColor }) => brandColor && css`
    fill: ${(props) => props.theme.brand};
  `}
  &.mediumUp {
    @media (max-width: 667px) {
      display: none;
    }
  }
`;

Icon.defaultProps = {
  theme: {
    brand: '#00BBEF',
  },
};

export const NoStyleButton = css`
  appearance: none;
  background: none;
  border-top: none;
  border-left: none;
  border-right: none;
  border-bottom: none;
  margin: ${(props) => props.margin};
  padding: ${(props) => props.padding};
  &:active,
  &:focus {
    outline: none;
  }
`;

export const GhostButton = styled.button`
  ${NoStyleButton}
  display: inline-flex;
  align-items: center;
  justify-content: center;
  svg {
    fill: ${(props) => props.theme.brand};
  }
  ${(props) => props.fill && css`
    svg {
      fill: ${props.fill};
    }
  `}
  &.mediumUp {
    @media (max-width: 667px) {
      display: none;
    }
  }
`;

GhostButton.defaultProps = {
  theme: {
    brand: '#00BBEF',
  },
};

export const TextButton = styled.button`
  color: ${(props) => props.color ? props.color : props.theme.brand};
  appearance: none;
  border: none;
  background: none;
  &:active,
  &:focus {
    outline: none;
  }
  &:hover {
    ${media.desktop`
      color: ${(props) => props.theme.darken};
    `}
  }
`;

TextButton.defaultProps = {
  theme: {
    brand: '#00BBEF',
    darken: '#1588AA',
  },
};
