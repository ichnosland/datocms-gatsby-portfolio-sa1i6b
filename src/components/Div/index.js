/**
*
* Div
*
*/

import styled, { css } from 'styled-components';
import media from 'style/mediainjector';

const Div = styled.div`
  display: ${(props) => props.display ? props.display : 'block'};
  position: ${(props) => props.position ? props.position : 'relative'};
  width: ${(props) => props.width};
  max-width: ${(props) => props.maxWidth ? props.maxWidth : '100%'};
  height: ${(props) => props.height};
  color: ${(props) => props.color ? props.color : 'inherit'};
  text-align: ${(props) => props.align};
  float: ${(props) => props.float ? props.float : 'none'};
  margin: ${(props) => props.margin ? props.margin : '0'};
  padding: ${(props) => props.padding ? props.padding : '0'};
  border: ${(props) => props.border ? props.border : 'none'};
  border-width: ${(props) => props.borderwidth};
  border-style: ${(props) => props.borderstyle};
  border-color: ${(props) => props.bordercolor};
  background-color: ${(props) => props.bgColor ? props.bgColor : 'transparent'};
  border-radius: ${(props) => props.radius};
  overflow: ${(props) => props.overflow ? props.overflow : 'visible'};
  user-select: ${(props) => props.userSelect};
  pointer-events: ${(props) => props.pointerEvents};
  z-index: ${(props) => props.zindex};
  ${(props) => props.nobreak && css`
    page-break-inside: avoid;
  `}
  ${media.print`
    ${({ noPrint }) => noPrint && css`
        display:none;
    `}
  `}
`;

export default Div;
