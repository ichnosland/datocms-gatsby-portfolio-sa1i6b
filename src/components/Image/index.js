/**
*
* Image
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import media from 'style/mediainjector';

export const ImageShell = styled.img`
  display: ${(props) => props.display};
  border: ${(props) => props.border};
  width: ${(props) => props.width};
  max-width: ${(props) => props.maxWidth};
  height: ${(props) => props.height};
  padding: ${(props) => props.padding};
  margin: ${(props) => props.margin};
  ${({ responsive }) => responsive && css`
    display: block;
    width: 100%;
    max-width: 100%;
  `}
  ${media.print`
    ${({ noPrint }) => noPrint && css`
        display: none;
    `}
  `}
`;

class Image extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    const {
      display,
      border,
      width,
      maxWidth,
      height,
      padding,
      margin,
      responsive,
      noPrint,
      src,
      alt,
      className,
    } = this.props;
    return (
      <ImageShell
        display={display}
        border={border}
        width={width}
        maxWidth={maxWidth}
        height={height}
        padding={padding}
        margin={margin}
        responsive={responsive}
        noPrint={noPrint}
        alt={alt}
        src={src}
        className={className}
      />
    );
  }
}

Image.propTypes = {
  display: PropTypes.string,
  border: PropTypes.string,
  width: PropTypes.string,
  maxWidth: PropTypes.string,
  height: PropTypes.string,
  padding: PropTypes.string,
  margin: PropTypes.string,
  responsive: PropTypes.bool,
  noPrint: PropTypes.bool,
  alt: PropTypes.string.isRequired,
  src: PropTypes.string.isRequired,
  className: PropTypes.string,
};

export default Image;
