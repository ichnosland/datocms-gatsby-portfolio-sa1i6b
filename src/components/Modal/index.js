/**
*
* Modal
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import Div from 'components/Div';
import FlexBox from 'components/FlexBox';
import Image from 'components/Image';
import TopBar from 'components/TopBar';
import { Button, Icon } from 'components/Button';
import media from 'style/mediainjector';
import buttonicon from 'icons/buttons';
import { colore } from 'style/color';

export const ModalOverlay = styled(FlexBox)`
  background-color: ${(props) => props.overlaybg ? props.overlaybg : props.theme.modalbg};
  position: fixed;
  top: 0;
  left: 0;
  right:0;
  bottom: 0;
  z-index: 16;
  overflow: auto;
  ${({ show }) => !show && css`
    display: none;
  `}
  ${({ topbar }) => topbar && css`
    align-items: flex-start;
    overflow: scroll;
  `}
`;

ModalOverlay.defaultProps = {
  theme: {
    modalbg: 'rgba(0, 0, 0, 0.75)',
  },
};

export const ModalContent = styled(Div)`
  max-width: ${(props) => props.maxWidth ? props.maxWidth : '860px'};
  background-color: ${(props) => props.bgcolor ? props.bgcolor : colore.ui.contrast};
  box-shadow: 0px 2px 11px -2px rgba(0,0,0,0.5);
  border-radius: 6px;
  ${media.mobile`
    margin: 10px;
  `}
  ${({ topbar }) => topbar && css`
    box-shadow: none;
  `}
  ${(props) => props.formBox && css`
    width: 100%;
    background-color: ${props.bgcolor ? props.bgcolor : props.theme.brand};
    color: ${props.color ? props.color : colore.ui.contrast};
    h2 {
      color: currentColor;
      margin-top: 0;
    }
    p {
      margin-bottom: 6px;
    }
    ${media.lt480`
      height: 100%;
      margin: 0;
      overflow-y: auto;
      border-radius: 0;
    `}
  `}
`;

ModalContent.defaultProps = {
  theme: {
    brand: '#00BBEF',
  },
};

export const ModalHeader = styled.h3`
  color: ${(props) => props.theme.brand};
  margin: 0;
  padding: 30px 30px 10px;
`;

ModalHeader.defaultProps = {
  theme: {
    brand: '#00BBEF',
  },
};

export const ModalFooter = styled(Div)`
  padding: 10px 30px 20px;
`;

export default class Modal extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    const {
      show,
      topbar,
      formBox,
      maxWidth,
      titolo,
      image,
      contenuto,
      overlaybg,
      bgcolor,
      color,
      barcolor,
      closeButton,
    } = this.props;
    return (
      <ModalOverlay show={show} overlaybg={overlaybg} topbar={topbar}>
        <ModalContent bgcolor={bgcolor} topbar={topbar} formBox={formBox} color={color} maxWidth={maxWidth}>
          {topbar &&
            <TopBar
              pinned
              title={titolo}
              single
              bgColor={barcolor}
              closeBtn={{
                url: '#',
                enabled: true,
                onClickFunction: closeButton.onClick,
              }}
            />
          }
          {titolo && !topbar && <ModalHeader>{titolo}</ModalHeader>}
          {image &&
            <Image src={image.src} widht={image.width} height={image.height} alt={image.alt} />
          }
          {contenuto}
          {closeButton && !topbar && (
            <ModalFooter>
              {closeButton && (
                <Button
                  actioncolor={closeButton.actionColor || ''}
                  onClick={closeButton.onClick}
                >
                  {closeButton.icon && <Icon {...buttonicon[closeButton.icon]} left />}
                  {closeButton.text || 'Chiudi'}
                </Button>
              )}
            </ModalFooter>
          )}
        </ModalContent>
      </ModalOverlay >
    );
  }
}

Modal.propTypes = {
  closeButton: PropTypes.shape({
    actioncolor: PropTypes.string,
    onClick: PropTypes.func.isRequired,
    icon: PropTypes.string,
  }),
  show: PropTypes.bool,
  topbar: PropTypes.bool,
  formBox: PropTypes.bool,
  maxWidth: PropTypes.string,
  image: PropTypes.object,
  contenuto: PropTypes.any,
  overlaybg: PropTypes.string,
  bgcolor: PropTypes.string,
  color: PropTypes.string,
  barcolor: PropTypes.string,
  titolo: PropTypes.string,
};
