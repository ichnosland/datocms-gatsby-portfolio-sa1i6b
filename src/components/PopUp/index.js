/**
*
* PopUp
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { ModalOverlay, ModalContent } from 'components/Modal';
import Flexbox from 'components/FlexBox';
import Image from 'components/Image';
import { Button } from 'components/Button';
import { colore } from 'style/color';
import media from 'style/mediainjector';
import Div from 'components/Div';
import HtmlStylesPopUp from 'style/HtmlStylesPopUp';

const innerMarginBottom = '20px';

export const PopUpContent = styled(ModalContent)`
  padding: 40px 30px 20px;
  max-width: 460px;
  min-width: 280px;
  ${media.lt667`
    max-width: 320px;
  `}
  img {
    display: block;
    margin: 0 auto ${innerMarginBottom};
  }
`;

export const PopUpHeader = styled.h4`
  color: ${(props) => props.theme.brand};
  text-align: center;
  margin: 0 auto ${innerMarginBottom};
`;

export const PopUpBody = styled(Div)`
  color: ${(props) => props.color ? props.color : colore.ui.txt};
  margin: 20px 0 30px;
  text-align: center;
  ${HtmlStylesPopUp}
`;

export const PopUpFooter = styled(Flexbox)`
  justify-content: center;
  margin: 30px 0 0;
  ${media.lt667`
    flex-wrap: wrap;
  `}
  button {
    flex: 1;
    border-radius: 4px;
    min-height: 42px;
    margin: 0;
    &:last-child {
      margin-left: 5px;
    }
    &:first-child {
      margin-right: 5px;
    }
    ${media.lt667`
      display: block;
      width: 100%;
      flex: auto;
      &:last-child {
        margin: 18px 0 0;
      }
      &:first-child {
        margin: 0;
      }
    `}
  }
`;

export default class PopUp extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    const {
      show,
      titolo,
      image,
      contenuto,
      bgcolor,
      closeButton,
      acceptButton,
    } = this.props;
    return (
      <ModalOverlay show={show}>
        <PopUpContent bgcolor={bgcolor}>
          {image &&
            <Image src={image.src} widht={image.width} height={image.height} alt={image.alt} />
          }
          {titolo && <PopUpHeader>{titolo}</PopUpHeader>}
          {contenuto && <PopUpBody
            dangerouslySetInnerHTML={{ // eslint-disable-line
              __html: contenuto,
            }}
          />}
          {(acceptButton || closeButton) && (
            <PopUpFooter>
              {closeButton && (
                <Button
                  outline={!!acceptButton}
                  onClick={closeButton.onClick}
                >
                  {closeButton.text || 'Annulla'}
                </Button>
              )}
              {acceptButton && (
                <Button
                  onClick={acceptButton.onClick}
                >
                  {acceptButton.text || 'Ok'}
                </Button>
              )}
            </PopUpFooter>
          )}
        </PopUpContent>
      </ModalOverlay >
    );
  }
}

PopUp.propTypes = {
  acceptButton: PropTypes.object,
  closeButton: PropTypes.object,
  show: PropTypes.bool,
  image: PropTypes.object,
  contenuto: PropTypes.string,
  bgcolor: PropTypes.string,
  titolo: PropTypes.string,
};
