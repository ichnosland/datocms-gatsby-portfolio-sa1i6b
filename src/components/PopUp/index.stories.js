import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, text, boolean, select, color } from '@storybook/addon-knobs';
import { ThemeProvider } from 'styled-components';
import 'global-styles';
import scrivi from 'images/infocard-icn_scrivi.png';
import gruppi from 'images/infocard-icn_gruppi.png';
import premium from 'images/infocard-icn_premium.png';
import blocchi from 'images/infocard-icn_blocchi.png';
import correzione from 'images/infocard-icn_correzione.png';
import checkmark from 'images/infocard-icn_checkmark.png';
import hourglass from 'images/infocard-icn_hourglass.png';
import surveydocenti from 'images/infocard-icn_survey-docenti.png';
import surveydownload from 'images/infocard-icn_survey-download.png';
import survey from 'images/infocard-icn_survey.png';
import { allThemes, themeOptions } from 'style/theme';
import PopUp from './index';

const imageList = {
  scrivi: {
    src: scrivi,
    width: '180px',
    height: '130px',
    alt: '',
  },
  gruppi: {
    src: gruppi,
    width: '180px',
    height: '130px',
    alt: '',
  },
  premium: {
    src: premium,
    width: '180px',
    height: '130px',
    alt: '',
  },
  blocchi: {
    src: blocchi,
    width: '180px',
    height: '130px',
    alt: '',
  },
  correzione: {
    src: correzione,
    width: '180px',
    height: '130px',
    alt: '',
  },
  checkmark: {
    src: checkmark,
    width: '180px',
    height: '130px',
    alt: '',
  },
  hourglass: {
    src: hourglass,
    width: '180px',
    height: '130px',
    alt: '',
  },
  surveydocenti: {
    src: surveydocenti,
    width: '180px',
    height: '130px',
    alt: '',
  },
  surveydownload: {
    src: surveydownload,
    width: '180px',
    height: '130px',
    alt: '',
  },
  survey: {
    src: survey,
    width: '180px',
    height: '130px',
    alt: '',
  },
};

storiesOf('PopUp', module)
  .addDecorator(withKnobs)
  .addWithJSX('PopUp', () => (
    <ThemeProvider theme={allThemes[select('theme', themeOptions, 'alatin')]}>
      <div>
        <PopUp
          show={boolean('show', true)}
          bgcolor={color('bgcolor')}
          titolo={text('Titolo', 'Titolo del PopUp')}
          contenuto={text('contenuto', 'Stai assegnando una nuova verifica')}
          acceptButton={boolean('acceptButton', true)}
          closeButton={boolean('closeButton', true)}
        />
      </div>
    </ThemeProvider>
  ), {
    skip: 2,
  })
  .addWithJSX('PopUp w/ image', () => (
    <ThemeProvider theme={allThemes[select('theme', themeOptions, 'alatin')]}>
      <div>
        <PopUp
          show
          image={imageList[select('image', Object.keys(imageList), 'scrivi')]}
          titolo="Titolo"
          contenuto={text('contenuto', 'Stai assegnando una nuova verifica')}
          closeButton={{ text: text('testo bottone', 'Ok') }}
        />
      </div>
    </ThemeProvider>
  ), {
    skip: 2,
  })
  .addWithJSX('PopUp w/ html', () => (
    <ThemeProvider theme={allThemes[select('theme', themeOptions, 'alatin')]}>
      <div>
        <PopUp
          show
          titolo="Html con css"
          contenuto={`
            <p>Questo è un paragrafo senza stili particolari applicati. Tanto per gradire</p>
            <p>Questo è un altro, per rendersi conto.</p>
            <ul>
              <li>questa una UL</li>
              <li>con qualche voce</li>
              <li>per testare</li>
            </ul>
            <ul class="fake-table">
              <li>questa una UL con classe 'fake-table'</li>
              <li>con qualche voce</li>
              <li>per testare</li>
            </ul>
          `}
          closeButton={{ text: text('testo bottone', 'Ok') }}
        />
      </div>
    </ThemeProvider>
  ), {
    skip: 2,
  });
