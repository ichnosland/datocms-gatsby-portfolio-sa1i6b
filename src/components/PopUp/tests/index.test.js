import React from 'react';
import { shallow } from 'enzyme';

import Div from 'components/Div';
import { ModalOverlay, ModalHeader } from 'components/Modal';
import { Button } from 'components/Button';
import Image from 'components/Image';
import scrivi from 'images/infocard-icn_scrivi.png';
import PopUp, {
  PopUpContent,
  PopUpBody,
  PopUpHeader,
  PopUpFooter,
} from '../index';


const popUpMock = {
  titolo: 'Questo è un titolo',
  contenuto: 'Questo è un contenuto',
  show: true,
  closeButton: {
    onClick: () => {},
  },
  enableOnClose: true,
};

describe('<PopUp />', () => {
  it('checks PopUp props with at least one button', () => {
    const renderedComponent = shallow(
      <PopUp
        {...popUpMock}
      />
    );
    expect(renderedComponent.find(PopUpContent).length).toBe(1);
    expect(renderedComponent.find(PopUpHeader).length).toBe(1);
    expect(renderedComponent.find(PopUpBody).length).toBe(1);
    expect(renderedComponent.find(PopUpFooter).length).toBe(1);
    expect(renderedComponent.find(Button).length).toBe(1);
  });

  it('checks PopUp props when no buttons provided', () => {
    const props = {
      ...popUpMock,
      closeButton: undefined,
    };
    const renderedComponent = shallow(
      <PopUp
        {...props}
      />
    );
    expect(renderedComponent.find(PopUpContent).length).toBe(1);
    expect(renderedComponent.find(PopUpHeader).length).toBe(1);
    expect(renderedComponent.find(PopUpBody).length).toBe(1);
    expect(renderedComponent.find(PopUpFooter).length).toBe(0);
  });

  it('checks PopUp props when acceptButton provided too', () => {
    const props = {
      ...popUpMock,
      acceptButton: {
        onClick: () => {},
      },
    };
    const renderedComponent = shallow(
      <PopUp
        {...props}
      />
    );
    expect(renderedComponent.find(PopUpContent).length).toBe(1);
    expect(renderedComponent.find(PopUpHeader).length).toBe(1);
    expect(renderedComponent.find(PopUpBody).length).toBe(1);
    expect(renderedComponent.find(PopUpFooter).length).toBe(1);
    expect(renderedComponent.find(Button).length).toBe(2);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('checks PopUp props when acceptButton provided too when label is provided', () => {
    const props = {
      ...popUpMock,
      acceptButton: {
        onClick: () => {},
        text: 'accept',
      },
      closeButton: {
        ...popUpMock.closeButton,
        text: 'cancella',
      },
    };
    const renderedComponent = shallow(
      <PopUp
        {...props}
      />
    );
    expect(renderedComponent.find(PopUpContent).length).toBe(1);
    expect(renderedComponent.find(PopUpHeader).length).toBe(1);
    expect(renderedComponent.find(PopUpBody).length).toBe(1);
    expect(renderedComponent.find(PopUpFooter).length).toBe(1);
    expect(renderedComponent.find(Button).length).toBe(2);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('checks PopUp props with empty title', () => {
    const renderedComponent = shallow(
      <PopUp
        {...{
          ...popUpMock,
          titolo: '',
        }}
      />
    );
    expect(renderedComponent.find(ModalHeader).length).toBe(0);
  });

  it('checks PopUp props with an image', () => {
    const renderedComponent = shallow(
      <PopUp
        {...{
          ...popUpMock,
          image: {
            src: scrivi,
            width: '180px',
            height: '130px',
            alt: 'Traduci',
          },
        }}
      />
    );
    expect(renderedComponent.find(Image).length).toBe(1);
  });

  it('checks PopUp props with empty content', () => {
    const renderedComponent = shallow(
      <PopUp
        {...{
          ...popUpMock,
          contenuto: '',
        }}
      />
    );
    expect(renderedComponent.find(PopUpBody).length).toBe(0);
  });

  it('checks ModalOverlay must be show=true when must be shown', () => {
    const renderedComponent = shallow(
      <PopUp
        {...{ ...popUpMock }}
      />
    );
    expect(renderedComponent.find(ModalOverlay).props().show).toBe(true);
  });

  it('checks ModalOverlay must be show=false when must not be shown', () => {
    const renderedComponent = shallow(
      <PopUp
        {...{ ...popUpMock, show: false }}
      />
    );
    expect(renderedComponent.find(ModalOverlay).props().show).toBe(false);
  });
});

describe('<PopUpHeader />', () => {
  it('should render a <h4> tag', () => {
    const renderedComponent = shallow(<PopUpHeader />);
    expect(renderedComponent.type()).toEqual('h4');
  });

  it('should render its css w/ props theme', () => {
    const theme = {
      brand: 'red',
    };
    const renderedComponent = shallow(<PopUpHeader theme={theme} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});

describe('<PopUpBody />', () => {
  it('should render a Div', () => {
    const renderedComponent = shallow(<PopUpBody />);
    expect(renderedComponent.find(Div).length).toEqual(1);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props color', () => {
    const color = 'yellow';
    const renderedComponent = shallow(<PopUpBody color={color} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
