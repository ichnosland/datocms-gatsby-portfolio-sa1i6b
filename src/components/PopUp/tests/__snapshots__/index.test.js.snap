// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PopUp /> checks PopUp props when acceptButton provided too 1`] = `
<Modal__ModalOverlay
  show={true}
  theme={
    Object {
      "modalbg": "rgba(0, 0, 0, 0.75)",
    }
  }
>
  <PopUp__PopUpContent>
    <PopUp__PopUpHeader>
      Questo è un titolo
    </PopUp__PopUpHeader>
    <PopUp__PopUpBody
      dangerouslySetInnerHTML={
        Object {
          "__html": "Questo è un contenuto",
        }
      }
    />
    <PopUp__PopUpFooter>
      <Button
        actioncolor=""
        onClick={[Function]}
        outline={true}
        theme={
          Object {
            "brand": "#00BBEF",
            "fonts": Object {
              "buttons": Object {
                "fontFamily": "inherit",
                "fontWeight": "inherit",
              },
            },
            "radius": Object {
              "buttons": "6px",
            },
          }
        }
      >
        Annulla
      </Button>
      <Button
        actioncolor=""
        onClick={[Function]}
        theme={
          Object {
            "brand": "#00BBEF",
            "fonts": Object {
              "buttons": Object {
                "fontFamily": "inherit",
                "fontWeight": "inherit",
              },
            },
            "radius": Object {
              "buttons": "6px",
            },
          }
        }
      >
        Ok
      </Button>
    </PopUp__PopUpFooter>
  </PopUp__PopUpContent>
</Modal__ModalOverlay>
`;

exports[`<PopUp /> checks PopUp props when acceptButton provided too when label is provided 1`] = `
<Modal__ModalOverlay
  show={true}
  theme={
    Object {
      "modalbg": "rgba(0, 0, 0, 0.75)",
    }
  }
>
  <PopUp__PopUpContent>
    <PopUp__PopUpHeader>
      Questo è un titolo
    </PopUp__PopUpHeader>
    <PopUp__PopUpBody
      dangerouslySetInnerHTML={
        Object {
          "__html": "Questo è un contenuto",
        }
      }
    />
    <PopUp__PopUpFooter>
      <Button
        actioncolor=""
        onClick={[Function]}
        outline={true}
        theme={
          Object {
            "brand": "#00BBEF",
            "fonts": Object {
              "buttons": Object {
                "fontFamily": "inherit",
                "fontWeight": "inherit",
              },
            },
            "radius": Object {
              "buttons": "6px",
            },
          }
        }
      >
        cancella
      </Button>
      <Button
        actioncolor=""
        onClick={[Function]}
        theme={
          Object {
            "brand": "#00BBEF",
            "fonts": Object {
              "buttons": Object {
                "fontFamily": "inherit",
                "fontWeight": "inherit",
              },
            },
            "radius": Object {
              "buttons": "6px",
            },
          }
        }
      >
        accept
      </Button>
    </PopUp__PopUpFooter>
  </PopUp__PopUpContent>
</Modal__ModalOverlay>
`;

exports[`<PopUpBody /> should render a Div 1`] = `
.c0 {
  color: rgb(78,78,78);
  margin: 20px 0 30px;
  text-align: center;
}

.c0 ul {
  text-align: left;
}

.c0 ul.fake-table {
  list-style: none;
  border: 1px solid rgba(0,0,0,0.2);
  padding: 0;
  border-radius: 6px;
}

.c0 ul.fake-table li {
  padding: 9px 14px;
  border-bottom: 1px solid rgba(0,0,0,0.1);
}

.c0 ul.fake-table li:last-child {
  border: none;
}

.c0 table {
  border-collapse: collapse;
  border-spacing: 0;
}

.c0 table th {
  background: rgb(231,236,236);
}

.c0 table td,
.c0 table th {
  padding: 0.62em;
  border: 1px solid rgb(231,236,236);
  vertical-align: middle;
}

.c0 .common-table-wrap {
  width: 100%;
  min-width: 100%;
  overflow-x: auto;
  border: 1px solid rgb(209,209,209);
  background-color: rgb(255,255,255);
}

.c0 .common-table {
  width: 100% !important;
  border: 1px solid rgb(209,209,209);
}

.c0 .common-table th {
  color: $darker-text-color;
  font-weight: 400;
  border: 1px solid rgb(209,209,209);
  font-size: 21px;
  padding: 9px 12px;
}

.c0 .common-table td {
  padding: 9px 12px;
  border: 1px solid rgb(209,209,209);
}

.c0 .common-table p {
  margin: 6px 0;
}

<Div
  className="c0"
/>
`;

exports[`<PopUpBody /> should render its css w/ props color 1`] = `
.c0 {
  color: yellow;
  margin: 20px 0 30px;
  text-align: center;
}

.c0 ul {
  text-align: left;
}

.c0 ul.fake-table {
  list-style: none;
  border: 1px solid rgba(0,0,0,0.2);
  padding: 0;
  border-radius: 6px;
}

.c0 ul.fake-table li {
  padding: 9px 14px;
  border-bottom: 1px solid rgba(0,0,0,0.1);
}

.c0 ul.fake-table li:last-child {
  border: none;
}

.c0 table {
  border-collapse: collapse;
  border-spacing: 0;
}

.c0 table th {
  background: rgb(231,236,236);
}

.c0 table td,
.c0 table th {
  padding: 0.62em;
  border: 1px solid rgb(231,236,236);
  vertical-align: middle;
}

.c0 .common-table-wrap {
  width: 100%;
  min-width: 100%;
  overflow-x: auto;
  border: 1px solid rgb(209,209,209);
  background-color: rgb(255,255,255);
}

.c0 .common-table {
  width: 100% !important;
  border: 1px solid rgb(209,209,209);
}

.c0 .common-table th {
  color: $darker-text-color;
  font-weight: 400;
  border: 1px solid rgb(209,209,209);
  font-size: 21px;
  padding: 9px 12px;
}

.c0 .common-table td {
  padding: 9px 12px;
  border: 1px solid rgb(209,209,209);
}

.c0 .common-table p {
  margin: 6px 0;
}

<Div
  className="c0"
  color="yellow"
/>
`;

exports[`<PopUpHeader /> should render its css w/ props theme 1`] = `
.c0 {
  color: red;
  text-align: center;
  margin: 0 auto 20px;
}

<h4
  className="c0"
/>
`;
